src/proposer/proposer.cpp

```diff
@@ -114,26 +183,7 @@ class ProposerImpl : public Proposer {
           }
           wallet_ext.GetProposerState().m_status = Status::IS_PROPOSING;
           wallet_ext.GetProposerState().m_number_of_search_attempts += 1;
```

Comment:
I think, the rest of this scope could go to `ActiveProposerImpl::GenerateBlock` (prior calling `proposer::GenerateBlock`), and then you can write elegantly on line 172 (https://github.com/dtr-org/unit-e/pull/1051/files#diff-8c0991a8037e36c5e8a5ab28694b40b5R172):

```C++
const std::shared_ptr<const CBlock> = GenerateBlock(...)
```

Reply:
I would rather not have ProposerState() specifics inside GenerateBlock(), this would actually mean having half of this logic inside there and half still in this function.